app/src/main/java/io/pumpkinz/pumpkinreader/model/Comment.java

```diff
@@ -13,13 +15,17 @@
     Comment parentComment;
     List<Comment> childComments;
 
-    public Comment() {}
+    public Comment() {
+        this.kids = new ArrayList<>();
+        this.childComments = new ArrayList<>();
```

Comment:
Good idea!
After all, the point of encapsulation is to not worry about the internal implementation.